### PR TITLE
Skip unresolved exceptions

### PR DESCRIPTION
**Pull request by haim-kermany**
## Description
handling uncouth exceptions

Please include a summary of the change and the issue that the PR is related to.
Please also include relevant motivation and context.
List any dependencies that are required for this change.

Related to # 172

## Type of Change

Please check the types of changes your PR introduces.

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Refactoring (non-breaking code restructuring that preserves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build-related update (CI workflow, test cases)
- [ ] Documentation update
- [ ] Other (please describe):

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so that it can be replicated.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tackle-test-generator-unit/src/org/konveyor/tackle/testgen/core/extender/TestSequenceExtender.java [hunk 1]
```diff
@@ -1217,6 +1217,18 @@ else if (SequenceUtil.isMapType(randoopType)) {
                 logger.warning(errmsg);
                 this.extSummary.uncovTestPlanRows__excp__NoSuchMethod++;
                 throw new RuntimeException(errmsg, nsme);
+            } catch (NonInstantiableTypeException nite){ // todo - Exception should be revisited
+                String errmsg = "type is not instrumented for type: " + paramType + "in signature "
```
Comment (rachelt44) [inline tackle-test-generator-unit/src/org/konveyor/tackle/testgen/core/extender/TestSequenceExtender.java:1221]: please fix: `type is not instrumented for type` to `type +paramType+ is not instantiable in signature`

File: tackle-test-generator-unit/src/org/konveyor/tackle/testgen/core/extender/TestSequenceExtender.java [hunk 2]
```diff
@@ -1805,6 +1817,10 @@ private void augmentClassSequencePool() {
                 this.extSummary.classNotFoundTypes.add(cnfe.getMessage());
             } catch (IllegalArgumentException iae) {
                 logger.warning("Error creating constructor sequence for " + clsName + ": " + iae.getMessage());
+            } catch (ClassCastException  cce) { // todo - Exception should be revisited
```
Comment (rachelt44) [inline tackle-test-generator-unit/src/org/konveyor/tackle/testgen/core/extender/TestSequenceExtender.java:1820]: please use one catch phrase with OR between it types` (IllegalArgumentException  | ClassCastException  | ExceptionInInitializerError  e)`

File: tackle-test-generator-unit/src/org/konveyor/tackle/testgen/model/JavaMethodModel.java [hunk 3]
```diff
@@ -217,6 +217,9 @@ private Set<Class<?>> getRelevantTypes(SootClass paramClass, Set<SootClass> clas
 			} catch (ClassNotFoundException | NoClassDefFoundError e) {
 				logger.warning(e.getMessage());
 				continue;
+			} catch (ClassFormatError e) { // todo - Exception should be revisited
```
Comment (rachelt44) [inline tackle-test-generator-unit/src/org/konveyor/tackle/testgen/model/JavaMethodModel.java:220]: please add to the previous catch phrase

File: tackle-test-generator-unit/src/org/konveyor/tackle/testgen/model/RapidTypeAnalysis.java [hunk 4]
```diff
@@ -113,9 +113,13 @@ private Set<String> performAnalysis(URLClassLoader urlLoader) {
...
 					}
-					
-					Body body = method.retrieveActiveBody();
-					
+					Body body = null;
+					try {
+						body = method.retrieveActiveBody();
+						//to figure out what is the problem
+					} catch (RuntimeException e) { // todo - Exception should be revisited
+						continue;
```
Comment (rachelt44) [inline tackle-test-generator-unit/src/org/konveyor/tackle/testgen/model/RapidTypeAnalysis.java:121]: please log a warning for this case - method body could not be retrieved

**Review by rachelt44 — CHANGES_REQUESTED** [review]

**Review by rachelt44 — APPROVED** [review]

File: tackle-test-generator-unit/src/org/konveyor/tackle/testgen/core/extender/TestSequenceExtender.java (hunk 1 above)
Comment (haim-kermany) [reply tackle-test-generator-unit/src/org/konveyor/tackle/testgen/core/extender/TestSequenceExtender.java]: done

File: tackle-test-generator-unit/src/org/konveyor/tackle/testgen/core/extender/TestSequenceExtender.java (hunk 2 above)
Comment (haim-kermany) [reply tackle-test-generator-unit/src/org/konveyor/tackle/testgen/core/extender/TestSequenceExtender.java]: done

File: tackle-test-generator-unit/src/org/konveyor/tackle/testgen/model/JavaMethodModel.java (hunk 3 above)
Comment (haim-kermany) [reply tackle-test-generator-unit/src/org/konveyor/tackle/testgen/model/JavaMethodModel.java]: done

File: tackle-test-generator-unit/src/org/konveyor/tackle/testgen/model/RapidTypeAnalysis.java (hunk 4 above)
Comment (haim-kermany) [reply tackle-test-generator-unit/src/org/konveyor/tackle/testgen/model/RapidTypeAnalysis.java:122]: done

**Pull request merged**